### ISSUE TITLE
Kill br process cause misleading error messages

### ISSUE THREAD

**Issue by aylei** [pingcap/br#1082]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

    * run br restore via the `Restore` CRD provided by tidb-operator
    * kill br process (`kill -9`)

2. What did you expect to see?

    * br process exited with error msg indicates that it was killed

3. What did you see instead?

    * br output lots of misleading error msgs

```
I0506 03:05:30.275369      11 restore.go:90] [2021/05/06 03:05:30.273 +00:00] [INFO] [coprocessor.go:1034] ["[TIME_COP_PROCESS] resp_time:485.46205ms txnStartTS:424743975747846146 region_id:25642 store_addr:db-tikv-2.db-tikv-peer.tidb1373933076654100628.svc:20160 kv_process_ms:483 scan_total_write:745919 scan_processed_write:745918 scan_total_data:0 scan_processed_data:0 scan_total_lock:1 scan_processed_lock:0"]
I0506 03:05:30.300859      11 restore.go:90]
I0506 03:05:30.345169      11 manager.go:274] reset cluster tidb1373933076654100628/nintendo-prod-390010 tikv_gc_life_time to 10m0s success
E0506 03:05:30.345231      11 manager.go:279] restore cluster tidb1373933076654100628/nintendo-prod-390010 from full failed, err: cluster tidb1373933076654100628/nintendo-prod-390010, wait pipe message failed, errMsg [2021/05/06 02:47:39.010 +00:00] [ERROR] [client.go:739] ["analyze table failed"] [db=tpcc] [table=warehouse] [table="{\"database_name\":\"\",\"table_name\":\"\",\"columns\":null,\"indices\":null,\"count\":0,\"modify_count\":0,\"partitions\":null}"] [error="[schema:1146]Table '.' doesn't exist"] [errorVerbose="[schema:1146]Table '.' doesn't exist\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/normalize.go:156\ngithub.com/pingcap/tidb/infoschema.(*infoSchema).TableByName\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/infoschema/infoschema.go:171\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).LoadStatsFromJSON\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/statistics/handle/dump.go:123\ngithub.com/pingcap/br/pkg/restore.(*Client).execChecksum\n\tgithub.com/pingcap/br@/pkg/restore/client.go:800\ngithub.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357"] [stack="github.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57"]
[2021/05/06 02:47:39.054 +00:00] [ERROR] [client.go:739] ["analyze table failed"] [db=tpcc] [table=item] [table="{\"database_name\":\"\",\"table_name\":\"\",\"columns\":null,\"indices\":null,\"count\":0,\"modify_count\":0,\"partitions\":null}"] [error="[schema:1146]Table '.' doesn't exist"] [errorVerbose="[schema:1146]Table '.' doesn't exist\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/normalize.go:156\ngithub.com/pingcap/tidb/infoschema.(*infoSchema).TableByName\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/infoschema/infoschema.go:171\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).LoadStatsFromJSON\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/statistics/handle/dump.go:123\ngithub.com/pingcap/br/pkg/restore.(*Client).execChecksum\n\tgithub.com/pingcap/br@/pkg/restore/client.go:800\ngithub.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357"] [stack="github.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57"]
[2021/05/06 02:55:09.102 +00:00] [ERROR] [client.go:739] ["analyze table failed"] [db=tpcc] [table=district] [table="{\"database_name\":\"\",\"table_name\":\"\",\"columns\":null,\"indices\":null,\"count\":0,\"modify_count\":0,\"partitions\":null}"] [error="[schema:1146]Table '.' doesn't exist"] [errorVerbose="[schema:1146]Table '.' doesn't exist\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/normalize.go:156\ngithub.com/pingcap/tidb/infoschema.(*infoSchema).TableByName\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/infoschema/infoschema.go:171\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).LoadStatsFromJSON\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/statistics/handle/dump.go:123\ngithub.com/pingcap/br/pkg/restore.(*Client).execChecksum\n\tgithub.com/pingcap/br@/pkg/restore/client.go:800\ngithub.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357"] [stack="github.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57"]
[2021/05/06 02:55:41.859 +00:00] [ERROR] [client.go:739] ["analyze table failed"] [db=tpcc] [table=new_order] [table="{\"database_name\":\"\",\"table_name\":\"\",\"columns\":null,\"indices\":null,\"count\":0,\"modify_count\":0,\"partitions\":null}"] [error="[schema:1146]Table '.' doesn't exist"] [errorVerbose="[schema:1146]Table '.' doesn't exist\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/normalize.go:156\ngithub.com/pingcap/tidb/infoschema.(*infoSchema).TableByName\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/infoschema/infoschema.go:171\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).LoadStatsFromJSON\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/statistics/handle/dump.go:123\ngithub.com/pingcap/br/pkg/restore.(*Client).execChecksum\n\tgithub.com/pingcap/br@/pkg/restore/client.go:800\ngithub.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357"] [stack="github.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57"]
, err: signal: killed
E0506 03:05:30.353595      11 restore_status_updater.go:79] Failed to update resotre [tidb1373933076654100628/nintendo-prod-390010], error: Operation cannot be fulfilled on restores.pingcap.com "nintendo-prod-390010": the object has been modified; please apply your changes to the latest version and try again
I0506 03:05:30.372803      11 restore_status_updater.go:76] Restore: [tidb1373933076654100628/nintendo-prod-390010] updated successfully
error: cluster tidb1373933076654100628/nintendo-prod-390010, wait pipe message failed, errMsg [2021/05/06 02:47:39.010 +00:00] [ERROR] [client.go:739] ["analyze table failed"] [db=tpcc] [table=warehouse] [table="{\"database_name\":\"\",\"table_name\":\"\",\"columns\":null,\"indices\":null,\"count\":0,\"modify_count\":0,\"partitions\":null}"] [error="[schema:1146]Table '.' doesn't exist"] [errorVerbose="[schema:1146]Table '.' doesn't exist\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/normalize.go:156\ngithub.com/pingcap/tidb/infoschema.(*infoSchema).TableByName\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/infoschema/infoschema.go:171\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).LoadStatsFromJSON\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/statistics/handle/dump.go:123\ngithub.com/pingcap/br/pkg/restore.(*Client).execChecksum\n\tgithub.com/pingcap/br@/pkg/restore/client.go:800\ngithub.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357"] [stack="github.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57"]
[2021/05/06 02:47:39.054 +00:00] [ERROR] [client.go:739] ["analyze table failed"] [db=tpcc] [table=item] [table="{\"database_name\":\"\",\"table_name\":\"\",\"columns\":null,\"indices\":null,\"count\":0,\"modify_count\":0,\"partitions\":null}"] [error="[schema:1146]Table '.' doesn't exist"] [errorVerbose="[schema:1146]Table '.' doesn't exist\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/normalize.go:156\ngithub.com/pingcap/tidb/infoschema.(*infoSchema).TableByName\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/infoschema/infoschema.go:171\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).LoadStatsFromJSON\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/statistics/handle/dump.go:123\ngithub.com/pingcap/br/pkg/restore.(*Client).execChecksum\n\tgithub.com/pingcap/br@/pkg/restore/client.go:800\ngithub.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357"] [stack="github.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57"]
[2021/05/06 02:55:09.102 +00:00] [ERROR] [client.go:739] ["analyze table failed"] [db=tpcc] [table=district] [table="{\"database_name\":\"\",\"table_name\":\"\",\"columns\":null,\"indices\":null,\"count\":0,\"modify_count\":0,\"partitions\":null}"] [error="[schema:1146]Table '.' doesn't exist"] [errorVerbose="[schema:1146]Table '.' doesn't exist\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/normalize.go:156\ngithub.com/pingcap/tidb/infoschema.(*infoSchema).TableByName\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/infoschema/infoschema.go:171\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).LoadStatsFromJSON\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/statistics/handle/dump.go:123\ngithub.com/pingcap/br/pkg/restore.(*Client).execChecksum\n\tgithub.com/pingcap/br@/pkg/restore/client.go:800\ngithub.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357"] [stack="github.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57"]
[2021/05/06 02:55:41.859 +00:00] [ERROR] [client.go:739] ["analyze table failed"] [db=tpcc] [table=new_order] [table="{\"database_name\":\"\",\"table_name\":\"\",\"columns\":null,\"indices\":null,\"count\":0,\"modify_count\":0,\"partitions\":null}"] [error="[schema:1146]Table '.' doesn't exist"] [errorVerbose="[schema:1146]Table '.' doesn't exist\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/normalize.go:156\ngithub.com/pingcap/tidb/infoschema.(*infoSchema).TableByName\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/infoschema/infoschema.go:171\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).LoadStatsFromJSON\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20201214152324-ce2f365189d3/statistics/handle/dump.go:123\ngithub.com/pingcap/br/pkg/restore.(*Client).execChecksum\n\tgithub.com/pingcap/br@/pkg/restore/client.go:800\ngithub.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357"] [stack="github.com/pingcap/br/pkg/restore.(*Client).GoValidateChecksum.func1.2\n\tgithub.com/pingcap/br@/pkg/restore/client.go:739\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57"]
, err: signal: killed
```

4. What version of BR and TiDB/TiKV/PD are you using?

all: v4.0.11

5. Operation logs
   - Please upload `br.log` for BR if possible
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible
   - Please upload `tikv-importer.log` from TiKV-Importer if possible
   - Other interesting logs

see above

**Comment by 3pointer** [comment]
Fixed in https://github.com/pingcap/br/pull/962

**Comment by tennix** [comment]
> Fixed in #962

Which version will contain this fix?

**Comment by 3pointer** [comment]
> > Fixed in #962
> 
> Which version will contain this fix?

It will be fixed in 4.0.13.
